distribution/lib/Standard/Base/0.0.0-dev/src/Data/Filter_Condition.enso

```diff
@@ -184,30 +186,38 @@ type Filter_Condition
     ## Converts a `Filter_Condition` condition into a predicate taking an
        element and returning a value indicating whether the element should be
        accepted by the filter.
+
+       The predicate can handle `Nothing` values in all cases. However, the
+       predicate will raise an error if the value is not of the expected type.
     to_predicate : (Any -> Boolean)
     to_predicate self = case self of
-        Less value -> <value
-        Equal_Or_Less value -> <=value
+        Less value -> handle_nothing (<value)
+        Equal_Or_Less value -> handle_nothing (<=value)
         Equal value -> ==value
-        Equal_Or_Greater value -> >=value
-        Greater value -> >value
+        Equal_Or_Greater value -> handle_nothing (>=value)
+        Greater value -> handle_nothing (>value)
         Not_Equal value -> !=value
-        Between lower upper -> elem ->
+        Between lower upper -> handle_nothing <| elem->
             (lower <= elem) && (elem <= upper)
-        Equal_Ignore_Case value locale -> elem-> elem.equals_ignore_case value locale
-        Starts_With prefix case_sensitivity -> _.starts_with prefix case_sensitivity
-        Ends_With suffix case_sensitivity -> _.ends_with suffix case_sensitivity
-        Contains substring case_sensitivity -> _.contains substring case_sensitivity
-        Not_Contains substring case_sensitivity -> v-> v.contains substring case_sensitivity . not
+        Equal_Ignore_Case value locale ->
+            handle_nothing <| txt-> (txt : Text).equals_ignore_case value locale
+        Starts_With prefix case_sensitivity ->
+            handle_nothing <| txt-> (txt : Text).starts_with prefix case_sensitivity
+        Ends_With suffix case_sensitivity ->
+            handle_nothing <| txt-> (txt : Text).ends_with suffix case_sensitivity
+        Contains substring case_sensitivity ->
+            handle_nothing <| txt-> (txt : Text).contains substring case_sensitivity
+        Not_Contains substring case_sensitivity ->
+            handle_nothing <| txt-> (txt : Text).contains substring case_sensitivity . not
```

Comment:
I added the `txt:Text` checks, because I think that `Expected txt to be Text, but got [...].` will be more readable than e.g. `Method `.equals_ignore_case` of type [...] could not be found.`.